ansible/cluster-setup/site.yaml

```diff
@@ -63,59 +51,91 @@
-  - name: Run Vault Config Chart
+    shell: "kubectl wait --for=condition=Ready pod/vault-0 -n vault --timeout=300s"
+  - name: Retrieve Vault Token
     with_items: "{{ env }}"
-    kubernetes.core.helm:
-      state: present
-      name: vault-config
-      namespace: "vault-admin"
-      chart_ref: ../../charts/vault-config
-      values:
-        git:
-          organization: "{{ lookup('ansible.builtin.env', 'GITHUB_ORGANIZATION') }}"
-          applicationId: "{{ lookup('ansible.builtin.env', 'GITHUB_APP_ID') }}"
-- name: Install RHSSO Operator Chart.
+    kubernetes.core.k8s_info:
+      api_version: v1
+      kind: Secret
+      name: vault-init
+      namespace: vault
+    register: vault_init
+  - name: Extract Vault Token
+    set_fact:
+      vault_token: "{{ vault_init.results[0].resources[0].data.root_token | b64decode }}"
+  - name: Populate Vault with Dev Spaces Secrets
+    with_items: "{{ env }}"
+    community.hashi_vault.vault_write:
+      url: "https://vault-vault.apps{{ cluster_host }}"
+      path: kv/{{ item.janusinstancename }}/dev-spaces
+      validate_certs: false
+      data:
+        id: "{{ lookup('ansible.builtin.env', 'GITHUB_DEV_SPACES_CLIENT_ID') }}"
+        secret: "{{ lookup('ansible.builtin.env', 'GITHUB_DEV_SPACES_CLIENT_SECRET') }}"
+      token: "{{ vault_token }}"
+  - name: Populate Vault with Github Credentials
+    with_items: "{{ env }}"
+    community.hashi_vault.vault_write:
+      url: "https://vault-vault.apps{{ cluster_host }}"
+      path: kv/{{ item.janusinstancename }}/github-plugin
+      validate_certs: false
+      data:
+        app_id: "{{ lookup('ansible.builtin.env', 'GITHUB_APP_ID') }}"
+        org: "{{ lookup('ansible.builtin.env', 'GITHUB_ORGANIZATION') }}"
+        private_key: "{{ lookup('ansible.builtin.file', lookup('ansible.builtin.env', 'GITHUB_KEY_FILE')) }}"
+      token: "{{ vault_token }}"
+- name: Install and Configure GitOps.
   hosts: 'localhost'
   tasks:
-  - name: Run RHSSO Helm Chart
+  - name: Install GitOps Helm Chart
     with_items: "{{ env }}"
     kubernetes.core.helm:
       state: present
-      name: rhsso-operator
-      namespace: "{{ item.namespace }}"
-      chart_ref: ../../charts/operator
-      values_files: ["../../charts/operator/values-rhsso-operator.yaml"]
+      name: argocd
+      namespace: "{{ item.gitopsnamespace }}"
+      chart_ref: ../../charts/gitops-operator
+      values:
+        namespaces:
+          - assemble-argocd
+          - infra-argocd
       create_namespace: true
-- name: Install Git Webhook Operator
-  hosts: 'localhost'
-  tasks:
-  - name: Run Git Webhook Operator Helm Chart
+  - name: Apply ApplicationSet for Infra Deployment
```

Comment:
It looks like the task "Apply ApplicationSet for Infra Deployment" is in there twice,  is this intentionaly and we're applying the applicationset chart twice?

Reply:
Removed duplicate